src/reducers/cartReducer.js

```diff
@@ -1,15 +1,33 @@
 export const initialCart = {
     cartItems: [],
-    totalAmount: 0
+    totalAmount: 0,
+    numOfItemsinCart: 0
 }
 
-function cartReducer(state, {type, payload}) {
+function cartReducer(state, { type, payload }) {
+    const itemPresent = state.cartItems.filter(item => item.id === payload.id)
```

Comment:
You can move this logic to the component itself and try to have two more actions like `INCREAMENT_QNT` & `DECREMENT_QNT` which will handle your quantity-related operations.

Reply:
cool, will update that